src/dbus_api/types.rs

```diff
@@ -132,6 +132,23 @@ impl ActionQueue {
         self.queue.push_back(DeferredAction::Remove(object_path))
     }
 
+    /// Push Remove actions for a path and all paths that reference it via
+    /// the OpContext.parent field.
+    pub fn push_remove_recursive(&mut self,
+                                 item: &Path<'static>,
+                                 tree: &Tree<MTFn<TData>, TData>)
+                                 -> () {
+        for opath in tree.iter() {
```

Comment:
Try instead:

```
let internal = |root: &Path<'static>| -> () {
    for opath in tree.iter().filter(|opath| opath.get_data().as_ref().map_or(false, |op_ctx| op_ctx.parent == *item) {
        internal(opath.get_name())
    }
    self.push_remove(root.clone());
};

internal(item);
```

The advantage is that you no longer appear to be recursing on the tree, which is nice, because that is not what is happening.

You may have to use a move, though.

Reply:
I have no objections to this suggestion, but am having some trouble getting it to work (i.e. compile). Any chance you could take a stab?